store/sqlstore/channel_store.go

```diff
@@ -3440,26 +3440,35 @@ func (s SqlChannelStore) CreateSidebarCategory(userId, teamId string, newCategor
 
 	defer finalizeTransaction(transaction)
 
-	sql, args, _ := s.getQueryBuilder().
-		Select("MAX(SortOrder)").
-		From("SidebarCategories").
-		GroupBy("SortOrder").
-		Where(sq.And{
-			sq.Eq{"TeamId": teamId},
-			sq.Eq{"UserId": userId},
-		}).ToSql()
-	maxOrder, err := transaction.SelectInt(sql, args...)
-	if err != nil {
-		return nil, model.NewAppError("SqlPostStore.CreateSidebarCategory", "store.sql_channel.sidebar_categories.app_error", nil, err.Error(), http.StatusInternalServerError)
+	categoriesWithOrder, appErr := s.getSidebarCategoriesT(transaction, userId, teamId)
+	if appErr != nil {
+		return nil, appErr
 	}
-	categoryId := model.NewId()
+	if len(categoriesWithOrder.Categories) < 1 {
```

Comment:
I was going to ask why we'd error out when you have no categories, but I guess this'll make it easier to not get into a weird cases by hitting this API before migration has occurred.